repair/repair.cc

```diff
@@ -47,9 +47,33 @@
 
 #include <seastar/core/gate.hh>
 #include <seastar/util/defer.hh>
+#include <seastar/core/metrics_registration.hh>
 
 logging::logger rlogger("repair");
 
+struct node_ops_metrics {
+    seastar::metrics::metric_groups _metrics;
+    uint64_t bootstrap_nr_ranges_total{0};
+    uint64_t bootstrap_nr_ranges_finished{0};
+    uint64_t rebuild_nr_ranges_total{0};
+    uint64_t rebuild_nr_ranges_finished{0};
+    node_ops_metrics() {
+        namespace sm = seastar::metrics;
+        _metrics.add_group("node_ops", {
+            sm::make_derive("bootstrap_nr_ranges_total", bootstrap_nr_ranges_total,
+                            sm::description("Number of total ranges for bootstrap operation on this shard.")),
+            sm::make_derive("bootstrap_nr_ranges_finished", bootstrap_nr_ranges_finished,
+                            sm::description("Number of finished ranges for bootstrap operation on this shard.")),
```

Comment:
I don't understand the difference between total and finish.
the nr in the middle is not clear.

A good metrics name's suffix should be the metrics unit, total typically used for a counter (or derived)

Reply:
There is no unit. The metric is the number of ranges. For example,  

    scylla_node_ops_bootstrap_nr_ranges_finished{shard="0",type="derive"} 50
    scylla_node_ops_bootstrap_nr_ranges_total{shard="0",type="derive"} 1040

"total" means we have 1040 ranges to work on.
"finished "means 50 out of 1040 ranges have been finished.

Reply:
so finish is going from 0 to `total` if the value is going up and down it should be a gauge.
Is total a static value, or will it changed over time?

Reply:
Yes, finished goes from 0 to total. Use sm::make_gauge?

total is a static value per operation, but total can change for different operations, e.g., rebuild after topology change, the number of ranges a node owns have changed. Looks like I need to use sm::make_gauge for total too?

Reply:
yes, both need to be gauge

Reply:
Ok. I will make them gauge.

Reply:
one more thing I've noticed the parameter references or function should be the last parameter, we support the older order, but it's nicer that way

Reply:
OK. I will change the order.